### ISSUE TITLE
Doesn't build with fluidsynth 2[.0.4]

### ISSUE THREAD

**Issue by mxmilkb** [tim-janik/beast#100]
```
bse/bsesoundfont.cc: In function ‘Bse::Error bse_sound_font_load_blob(BseSoundFont*, BseStorage::BlobP, gboolean)’:
bse/bsesoundfont.cc:138:19: error: aggregate ‘fluid_preset_t fluid_preset’ has incomplete type and cannot be defined
    fluid_preset_t fluid_preset;
                   ^~~~~~~~~~~~
bse/bsesoundfont.cc:140:15: error: invalid use of incomplete type ‘fluid_sfont_t’ {aka ‘struct _fluid_sfont_t’}
    fluid_sfont->iteration_start (fluid_sfont);
               ^~
In file included from /usr/include/fluidsynth.h:95,
                 from bse/bsesoundfontrepo.hh:6,
                 from bse/bsesoundfont.cc:3:
/usr/include/fluidsynth/types.h:40:16: note: forward declaration of ‘fluid_sfont_t’ {aka ‘struct _fluid_sfont_t’}
 typedef struct _fluid_sfont_t fluid_sfont_t;                    /**< SoundFont */
                ^~~~~~~~~~~~~~
bse/bsesoundfont.cc:141:22: error: invalid use of incomplete type ‘fluid_sfont_t’ {aka ‘struct _fluid_sfont_t’}
    while (fluid_sfont->iteration_next (fluid_sfont, &fluid_preset))
                      ^~
In file included from /usr/include/fluidsynth.h:95,
                 from bse/bsesoundfontrepo.hh:6,
                 from bse/bsesoundfont.cc:3:
/usr/include/fluidsynth/types.h:40:16: note: forward declaration of ‘fluid_sfont_t’ {aka ‘struct _fluid_sfont_t’}
 typedef struct _fluid_sfont_t fluid_sfont_t;                    /**< SoundFont */
                ^~~~~~~~~~~~~~
make: *** [config-uname.mk:84: out/bse/bsesoundfont.o] Error 1

```

**Comment by tim-janik** [comment]
Libfluidsynth 2 is API incompatible with 1 in some places.
Updating Beast to build against 2 is on Stefans todo list (@swesterfeld ).

**Comment by swesterfeld** [comment]
I've ported the code to fluidsynth >= 2.0.0 and submitted the required changes here: https://github.com/tim-janik/beast/pull/101

**Comment by swesterfeld** [comment]
Building against fluidsynth 2 is fixed now https://github.com/tim-janik/beast/commit/4c65148cf152572cd0e7a8e012cd71f4216cda6b

**Comment by tim-janik** [comment]
Fixed with 4c65148cf152572cd0e7a8e012cd71f4216cda6b
